src/main/java/com/rapid7/client/dcerpc/msvcctl/ServiceControlManagerService.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2017, Rapid7, Inc.
+ *
+ * License: BSD-3-clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ */
+package com.rapid7.client.dcerpc.msvcctl;
+
+import com.rapid7.client.dcerpc.mserref.SystemErrorCode;
+import com.rapid7.client.dcerpc.msrrp.messages.HandleResponse;
+import com.rapid7.client.dcerpc.msrrp.objects.ContextHandle;
+import com.rapid7.client.dcerpc.msvcctl.messages.RChangeServiceConfigWRequest;
+import com.rapid7.client.dcerpc.msvcctl.messages.RChangeServiceConfigWResponse;
+import com.rapid7.client.dcerpc.msvcctl.messages.RControlServiceRequest;
+import com.rapid7.client.dcerpc.msvcctl.messages.ROpenSCManagerWRequest;
+import com.rapid7.client.dcerpc.msvcctl.messages.ROpenServiceWRequest;
+import com.rapid7.client.dcerpc.msvcctl.messages.RQueryServiceStatusRequest;
+import com.rapid7.client.dcerpc.msvcctl.messages.RQueryServiceStatusResponse;
+import com.rapid7.client.dcerpc.msvcctl.messages.RStartServiceWRequest;
+import com.rapid7.client.dcerpc.msvcctl.objects.RQueryServiceStatusInfo;
+import com.rapid7.client.dcerpc.objects.EmptyResponse;
+import com.rapid7.client.dcerpc.transport.RPCTransport;
+import java.io.IOException;
+
+public class ServiceControlManagerService
+{
+
+    public final static String REMOTE_REGISTRY = "Remoteregistry";
+    public final static int FULL_ACCESS = 0x000F003F;
+
+    private final RPCTransport transport;
+    private ContextHandle serviceManagerHandle;
```

Comment:
Would it make more sense for the service manager handle to be managed by the caller instead of persisting it in the Service?

Reply:
That's what I was thinking too.

Reply:
This is the same issue as the PolicyContextHandle; let's keep a consistent approach across all our RPC calls. Options are
1. Open handle -> Call service -> Close Handle (Do we have to do multiple un-necessary opening and closing handles for multiple calls?)
2. Maintain handle in service (Is there an issue with not closing this handle? What about multi threaded operations?)
3. Allow handle to be passed into service by caller. (Does this make the calling code more complex than necessary?)

Reply:
I've been passing handle into the methods. The service has a method to close the handle.

Reply:
I wouldn't worry too much about keeping it consistent between Service implementations since the use case of ContextHandle can be different in each service.

Reply:
Ok, changed code to pass around contextHandle